src/include/log.h

```diff
@@ -13,32 +34,43 @@
 /* Logging subsystem declarations. */
 #define	WT_LOG_ALIGN			128
 
+/*
+ * Atomically set the two components of the LSN.
+ */
+#define	lsn_file		u.l.file
+#define	lsn_offset		u.l.offset
+#define	WT_SET_LSN(l, f, o) (l)->u.file_offset = (((uint64_t)(f) << 32) + (o))
```

Comment:
The whole point is to set the LSN as a single 64-bit value rather than two assignments so that reading an LSN never sees something half-assigned.